include/boost/geometry/algorithms/detail/partition.hpp

```diff
@@ -84,23 +92,34 @@ inline void divide_into_subsets(Box const& lower_box,
     }
 }
 
-template <typename ExpandPolicy, typename Box, typename InputCollection>
+template
+<
+    typename ExpandPolicy,
+    typename Box,
+    typename ForwardRange,
+    typename IteratorVector
+>
 inline void expand_with_elements(Box& total,
-                InputCollection const& collection,
-                index_vector_type const& input)
+                ForwardRange const& /*forward_range*/,
```

Comment:
OK, I see. For Expand I don't think it will be useful, so IMO you can remove it - we can always readd it if really needed